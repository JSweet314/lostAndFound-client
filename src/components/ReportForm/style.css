.report-form {
  display: grid;
  width: 320px;
  grid-template-columns: 12rem 1fr;
  grid-template-rows: auto 1fr auto 1fr 1fr auto;
  grid-gap: 2rem;
  color: #fff;
  padding: 1rem 0 0 1rem;
}

.report-form__heading {
  grid-column: 1/3;
  font-size: 3rem;
  margin: 0;
  color: #fff;
}

.report-form__label {
  font-size: 2rem;
  font-weight: 800;
}

.report-form__textarea,
.report-form__input {
  height: 3rem;
  background-color: rgba(256, 256, 256, 0.5);
  font-family: 'Exo', sans-serif;
  font-size: 1.5rem;
  border: none;
  padding: 0.5rem;
}

.report-form__textarea {
  height: 6rem;
}

.report-form__textarea::placeholder,
.report-form__input::placeholder {
  font-family: 'Exo', sans-serif;
  color: rgba(256, 256, 256, 0.5)
}

.report-form__btn {
  grid-column: 1/3;
  padding: 1rem;
  border: 2px solid #625049;
  background-color: #C28744;
  color: #310906;
  font-size: 2rem;
  font-family: 'Exo', sans-serif;
  text-decoration: none;
  border-radius: 15px;
  box-shadow: 2px 2px 10px #310906
}

.report-form__btn:hover,
.report-form__submit:hover {
  background-color: #625049;
  color: #EBC9AD;
}