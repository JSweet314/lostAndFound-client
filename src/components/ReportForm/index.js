import React from 'react';
import PropTypes from 'prop-types';
import './style.css';
import { Link } from 'react-router-dom';

const ReportForm = ({
  name,
  description,
  date,
  reward,
  handleOnSubmit,
  handleOnChange,
  handleGoBack,
  routeId
}) => {
  const display = { display: routeId === 'lost' ? 'initial' : 'none' };
  const isDisabled = name && description && date ? false : true;
  return (
    <div className='report-form-container'>
      <form
        onSubmit={event => handleOnSubmit(event)}
        className='report-form'>
        <h2 className='report-form__heading'>{routeId}</h2>
        <label htmlFor='name' className='report-form__label'>Item Name</label>
        <input
          required
          onChange={event => handleOnChange(event)}
          value={name}
          name='name'
          placeholder='required'
          className='report-form__input name'
          type='text' />
        <label
          htmlFor='description'
          className='report-form__label'>
          Description
        </label>
        <textarea
          required
          onChange={event => handleOnChange(event)}
          value={description}
          name='description'
          placeholder='required'
          className='report-form__textarea'
          type='text' />
        <label htmlFor='date' className='report-form__label'>Date</label>
        <input
          required
          onChange={event => handleOnChange(event)}
          value={date}
          name='date'
          placeholder='required'
          className='report-form__input'
          type='date' />
        <label
          htmlFor='reward'
          style={display}
          className='report-form__label'>
          Reward?
        </label>
        <input
          onChange={event => handleOnChange(event)}
          value={reward}
          name='reward'
          style={display}
          placeholder='$?'
          className='report-form__input'
          type='text' />
        <div className='report-form__nav'>
          <button
            className='report-form__btn'
            onClick={event => handleGoBack(event)}>
            Back
          </button>
          <Link
            to='/report/map'
            disabled={isDisabled}
            type='submit'
            className='report-form__btn'>
            next
          </Link>
        </div>
      </form>
    </div>
  );
};

ReportForm.propTypes = {
  name: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  reward: PropTypes.string.isRequired,
  routeId: PropTypes.string.isRequired,
  handleOnSubmit: PropTypes.func.isRequired,
  handleOnChange: PropTypes.func.isRequired,
  handleGoBack: PropTypes.func.isRequired
};

export default ReportForm;