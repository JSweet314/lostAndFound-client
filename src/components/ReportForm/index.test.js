import React from 'react';
import { shallow } from 'enzyme';
import ReportForm from './index';

describe('ReportForm', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <ReportForm
        name=''
        description=''
        date=''
        location={{}}
        reward=''
        routeId='lost'
        handleGoBack={jest.fn()}
        handleOnChange={jest.fn()}
        handleOnSubmit={jest.fn()}
      />
    );
  });
  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match a snapshot for a found item route', () => {
    wrapper = shallow(
      <ReportForm
        name=''
        description=''
        date=''
        location={{}}
        reward=''
        routeId='found'
        handleGoBack={jest.fn()}
        handleOnChange={jest.fn()}
        handleOnSubmit={jest.fn()}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});