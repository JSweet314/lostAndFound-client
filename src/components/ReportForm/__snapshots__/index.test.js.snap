// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportForm should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportForm
    date=""
    description=""
    handleGoBack={[Function]}
    handleOnChange={[Function]}
    handleOnSubmit={[Function]}
    location=""
    name=""
    reward=""
    routeId="lost"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="report-form__heading"
>
          lost
</h2>,
        <label
          className="report-form__label"
          htmlFor="name"
>
          Item Name
</label>,
        <input
          className="report-form__input"
          name="name"
          onChange={[Function]}
          placeholder="required"
          required={true}
          type="text"
          value=""
/>,
        <label
          className="report-form__label"
          htmlFor="description"
>
          Description
</label>,
        <textarea
          className="report-form__textarea"
          name="description"
          onChange={[Function]}
          placeholder="required"
          required={true}
          type="text"
          value=""
/>,
        <label
          className="report-form__label"
          htmlFor="location"
>
          Last Known Location
</label>,
        <input
          className="report-form__input"
          name="location"
          onChange={[Function]}
          placeholder="required"
          required={true}
          type="text"
          value=""
/>,
        <label
          className="report-form__label"
          htmlFor="date"
>
          Date
</label>,
        <input
          className="report-form__input"
          name="date"
          onChange={[Function]}
          placeholder="required"
          required={true}
          type="date"
          value=""
/>,
        <label
          className="report-form__label"
          htmlFor="reward"
          style={
                    Object {
                              "display": "initial",
                            }
          }
>
          Reward?
</label>,
        <input
          className="report-form__input"
          name="reward"
          onChange={[Function]}
          placeholder="$?"
          style={
                    Object {
                              "display": "initial",
                            }
          }
          type="text"
          value=""
/>,
        <button
          className="report-form__btn"
          type="submit"
>
          Submit
</button>,
        <button
          className="report-form__btn"
          onClick={[Function]}
>
          Back
</button>,
      ],
      "className": "report-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "lost",
          "className": "report-form__heading",
        },
        "ref": null,
        "rendered": "lost",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Item Name",
          "className": "report-form__label",
          "htmlFor": "name",
        },
        "ref": null,
        "rendered": "Item Name",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "report-form__input",
          "name": "name",
          "onChange": [Function],
          "placeholder": "required",
          "required": true,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Description",
          "className": "report-form__label",
          "htmlFor": "description",
        },
        "ref": null,
        "rendered": "Description",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "report-form__textarea",
          "name": "description",
          "onChange": [Function],
          "placeholder": "required",
          "required": true,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Last Known Location",
          "className": "report-form__label",
          "htmlFor": "location",
        },
        "ref": null,
        "rendered": "Last Known Location",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "report-form__input",
          "name": "location",
          "onChange": [Function],
          "placeholder": "required",
          "required": true,
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Date",
          "className": "report-form__label",
          "htmlFor": "date",
        },
        "ref": null,
        "rendered": "Date",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "report-form__input",
          "name": "date",
          "onChange": [Function],
          "placeholder": "required",
          "required": true,
          "type": "date",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reward?",
          "className": "report-form__label",
          "htmlFor": "reward",
          "style": Object {
            "display": "initial",
          },
        },
        "ref": null,
        "rendered": "Reward?",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "report-form__input",
          "name": "reward",
          "onChange": [Function],
          "placeholder": "$?",
          "style": Object {
            "display": "initial",
          },
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
          "className": "report-form__btn",
          "type": "submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Back",
          "className": "report-form__btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Back",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="report-form__heading"
>
            lost
</h2>,
          <label
            className="report-form__label"
            htmlFor="name"
>
            Item Name
</label>,
          <input
            className="report-form__input"
            name="name"
            onChange={[Function]}
            placeholder="required"
            required={true}
            type="text"
            value=""
/>,
          <label
            className="report-form__label"
            htmlFor="description"
>
            Description
</label>,
          <textarea
            className="report-form__textarea"
            name="description"
            onChange={[Function]}
            placeholder="required"
            required={true}
            type="text"
            value=""
/>,
          <label
            className="report-form__label"
            htmlFor="location"
>
            Last Known Location
</label>,
          <input
            className="report-form__input"
            name="location"
            onChange={[Function]}
            placeholder="required"
            required={true}
            type="text"
            value=""
/>,
          <label
            className="report-form__label"
            htmlFor="date"
>
            Date
</label>,
          <input
            className="report-form__input"
            name="date"
            onChange={[Function]}
            placeholder="required"
            required={true}
            type="date"
            value=""
/>,
          <label
            className="report-form__label"
            htmlFor="reward"
            style={
                        Object {
                                    "display": "initial",
                                  }
            }
>
            Reward?
</label>,
          <input
            className="report-form__input"
            name="reward"
            onChange={[Function]}
            placeholder="$?"
            style={
                        Object {
                                    "display": "initial",
                                  }
            }
            type="text"
            value=""
/>,
          <button
            className="report-form__btn"
            type="submit"
>
            Submit
</button>,
          <button
            className="report-form__btn"
            onClick={[Function]}
>
            Back
</button>,
        ],
        "className": "report-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "lost",
            "className": "report-form__heading",
          },
          "ref": null,
          "rendered": "lost",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Item Name",
            "className": "report-form__label",
            "htmlFor": "name",
          },
          "ref": null,
          "rendered": "Item Name",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "report-form__input",
            "name": "name",
            "onChange": [Function],
            "placeholder": "required",
            "required": true,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Description",
            "className": "report-form__label",
            "htmlFor": "description",
          },
          "ref": null,
          "rendered": "Description",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "report-form__textarea",
            "name": "description",
            "onChange": [Function],
            "placeholder": "required",
            "required": true,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "textarea",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Last Known Location",
            "className": "report-form__label",
            "htmlFor": "location",
          },
          "ref": null,
          "rendered": "Last Known Location",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "report-form__input",
            "name": "location",
            "onChange": [Function],
            "placeholder": "required",
            "required": true,
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Date",
            "className": "report-form__label",
            "htmlFor": "date",
          },
          "ref": null,
          "rendered": "Date",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "report-form__input",
            "name": "date",
            "onChange": [Function],
            "placeholder": "required",
            "required": true,
            "type": "date",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reward?",
            "className": "report-form__label",
            "htmlFor": "reward",
            "style": Object {
              "display": "initial",
            },
          },
          "ref": null,
          "rendered": "Reward?",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "report-form__input",
            "name": "reward",
            "onChange": [Function],
            "placeholder": "$?",
            "style": Object {
              "display": "initial",
            },
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
            "className": "report-form__btn",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Back",
            "className": "report-form__btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Back",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
